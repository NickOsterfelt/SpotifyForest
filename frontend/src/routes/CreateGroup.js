import React, { useEffect, useState } from "react";
import Api from "../engine/Api";
import CreateGroupForm from "../components/CreateGroupForm";

/**
 * CreateGroup component is the route for creating new groups
 * It displays the CreateGroupForm, and handles the data associated
 * with the form and telling the backend to create a new group.
 */
function CreateGroup(props) {
    const [formData, setFormData] = useState({name: "", info: ""});
    const [loading, setLoading] = useState(false);

    useEffect(() => {
        async function createGroup() {
            if (loading) {
                setLoading(false);
                let userId = localStorage.getItem("userId")
                await Api.createGroup({ data: formData, userId: userId });
                props.history.push("/");
            }
        }
        createGroup();
    }, [loading, formData]);

    const handleChange = (evt) => {
        const { name, value } = evt.target;
        setFormData(data => ({ ...data, [name]: value }));
    }

    const handleSubmit = (evt) => {
        evt.preventDefault();
        setLoading(true);
    }

    return (
        <div className="container mt-5">
            <div className="row d-flex justify-content-center">
                <div className="col-6 bg-dark text-left text-light rounded p-5">
                    <CreateGroupForm
                        formData={formData}
                        handleChange={handleChange}
                        handleSubmit={handleSubmit}
                    />
                </div>
            </div>

        </div>
    )
}

export default CreateGroup;
