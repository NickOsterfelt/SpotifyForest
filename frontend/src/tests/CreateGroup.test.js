import React from "react";
import { render } from "@testing-library/react";
import CreateGroup from "../routes/CreateGroup";

it("renders without crashing", function() {
  render(<CreateGroup />);
});

it("matches snapshot", function() {
  const { asFragment } = render(<CreateGroup />);
  expect(asFragment()).toMatchSnapshot();
});