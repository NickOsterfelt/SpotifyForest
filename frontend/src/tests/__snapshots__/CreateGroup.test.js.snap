// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container mt-5"
  >
    <div
      class="row d-flex justify-content-center"
    >
      <div
        class="col-6 bg-dark text-left text-light rounded p-5"
      >
        <form
          class=""
        >
          <div
            class="form-group"
          >
            <label
              class="form-label"
              for="groupName"
            >
              Group Name
            </label>
            <input
              class="form-control"
              id="groupName"
              name="name"
              placeholder="Enter a name for the group"
              type="text"
              value=""
            />
          </div>
          <div
            class="form-group"
          >
            <label
              class="form-label"
              for="groupInfo"
            >
              Description
            </label>
            <input
              class="form-control"
              id="groupInfo"
              name="info"
              placeholder="What should people know about the group?"
              type="text"
              value=""
            />
          </div>
          <button
            class="btn btn-success"
            type="submit"
          >
            Submit
          </button>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
